tls/s2n_psk.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#pragma once
+
+#include <s2n.h>
+
+#include "crypto/s2n_hash.h"
+#include "utils/s2n_blob.h"
+#include "utils/s2n_result.h"
+
+typedef enum {
+    S2N_PSK_TYPE_UNKNOWN_TYPE = 0,
+    S2N_PSK_TYPE_EXTERNAL,
+    S2N_PSK_TYPE_RESUMPTION
+} s2n_psk_type;
+
+struct s2n_psk {
+    s2n_psk_type type;
+    struct s2n_blob identity;
+    struct s2n_blob secret;
+    s2n_hash_algorithm hash_alg;
+    uint32_t obfuscated_ticket_age;
+    struct s2n_blob early_secret;
```

Comment:
Q: Is the `early_secret` referring to the `binder_key` derived from the PSK?  Can we be more specific and call it `binder_secret` or `early_binder_secret`?

Reply:
This is just the early secret, not binder-related. The early secret is separate from the binder secret, as show in the key schedule: https://tools.ietf.org/html/rfc8446#section-7.1 We're storing it so we can reuse it when calculating the derived secret later.

Reply:
Resolved in offline discussion, Thank-you.